testing/e2e/skr-tester/pkg/command/update.go

```diff
@@ -0,0 +1,157 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	broker "skr-tester/pkg/broker"
+	"skr-tester/pkg/logger"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type UpdateCommand struct {
+	cobraCmd          *cobra.Command
+	log               logger.Logger
+	instanceID        string
+	planID            string
+	updateMachineType bool
+	// TODO
+	updateOIDC bool
+}
+
+func NewUpdateCommand() *cobra.Command {
+	cmd := UpdateCommand{}
+	cobraCmd := &cobra.Command{
+		Use:     "update",
+		Aliases: []string{"u"},
+		Short:   "Update the instnace",
+		Long:    "Update the instnace",
+		Example: "skr-tester update -i instanceID -p planID --updateMachineType                            Update the instance with new machineType.",
+
+		PreRunE: func(_ *cobra.Command, _ []string) error { return cmd.Validate() },
+		RunE:    func(_ *cobra.Command, _ []string) error { return cmd.Run() },
+	}
+	cmd.cobraCmd = cobraCmd
+
+	cobraCmd.Flags().StringVarP(&cmd.instanceID, "instanceID", "i", "", "InstanceID of the specific instance.")
+	cobraCmd.Flags().StringVarP(&cmd.planID, "planID", "p", "", "PlanID of the specific instance.")
+	cobraCmd.Flags().BoolVarP(&cmd.updateMachineType, "updateMachineType", "m", false, "Should update machineType.")
+
+	return cobraCmd
+}
+
+func (cmd *UpdateCommand) Run() error {
+	cmd.log = logger.New()
+	brokerClient := broker.NewBrokerClient(broker.NewBrokerConfig())
+	catalog, err := brokerClient.GetCatalog()
+	if err != nil {
+		return fmt.Errorf("failed to get catalog: %v", err)
+	}
+	services, ok := catalog["services"].([]interface{})
+	if !ok {
+		return errors.New("services field not found or invalid in catalog")
+	}
+	for _, service := range services {
+		serviceMap, ok := service.(map[string]interface{})
+		if !ok {
+			return errors.New("service is not a map[string]interface{}")
+		}
+		if cmd.updateMachineType {
+			currentMachineType, err := getCurrentMachineType(cmd.instanceID)
+			if err != nil {
+				return fmt.Errorf("failed to get current machine type: %v", err)
+			}
+			fmt.Printf("Current machine type: %s\n", *currentMachineType)
+			if serviceMap["id"] != "47c9dcbf-ff30-448e-ab36-d3bad66ba281" {
+				continue
+			}
+			plans, ok := serviceMap["plans"].([]interface{})
+			if !ok {
+				return errors.New("plans field not found or invalid in serviceMap")
+			}
+			for _, p := range plans {
+				planMap, ok := p.(map[string]interface{})
+				if !ok || planMap["id"] != cmd.planID {
+					continue
+				}
+				updateParams, err := extractUpdateParams(planMap)
+				if err != nil {
+					return fmt.Errorf("failed to extract update parameters: %v", err)
+				}
+				if len(updateParams) < 2 {
+					continue
+				}
+				for i, m := range updateParams {
+					if m == *currentMachineType {
+						newMachineType := updateParams[(i+1)%len(updateParams)].(string)
```

Comment:
This looks nice for a flow when the command is run without providing any machine type as a parameter to the command. I think it may be worthwhile to implement the flow when machine type name is provided as a parameter.

Reply:
I will add your idea to the ticket as an AC if you agree to wait with this feature. I would like to get the tests running as they are now, so we can get rid of nodejs.

Reply:
Sure, we can add extra functionality in next PRs.

Reply:
https://github.com/kyma-project/kyma-environment-broker/issues/1588#issuecomment-2565442026